File: Coding_Platforms/CodeChef/Practice/Golang/max_number.go
package main

import (
	"fmt"
)

func main() {

	//create variables
	var testcase, size int
	var data string

	fmt.Scan(&testcase) //take input for number of testcases

	for i := 0; i < testcase; i++ { //enter value for each test case

		fmt.Scan(&size)
		fmt.Scan(&data)

		var result int //create variable to store variable

		list := []rune(data) //add test case data into slice

		for idx, val := range list { //iterate over slice

			if idx == 0 { //if value is first

				if val == 49 || len(list) == 1 { // if first number is 1 or size of list is 1 then add 1 in result
					result++
				}
				continue

			} else if list[idx-1] == 48 && val == 49 { //if previous value is 0 and current value is 1 then add 1 to result
				result++

			} else if idx == len(list)-1 && val == 48 { //if last value is 0 then add 1 to result
				result++
			}
		}
		fmt.Println(result)
	}

}

File: Coding_Platforms/CodeChef/Practice/Python/max_number.py
def MaxNumber(test_data): # create function to check max number

    result = 0
    for k,v in enumerate(test_data): # iterate over list 

        if k == 0: # if value is first

            if (v or len(test_data)) == 1: # if value and size of list is 1 then add 1 to result
                result+=1

        elif test_data[k-1] == 0 and v == 1: # if previous value is 0 and current value is 1 then add 1 in result
            result +=1

        elif k == len(test_data)-1 and v == 0: # if last value is 0 then add 1 to result
            result+=1

    return result

for i in range(int(input())):# iterate over test case

    # take test size
    test_size = int(input())

    # Put list in MaxNumber function then print max number
    print(MaxNumber(list(map(int,input().strip()))))

